axon-server-connector/src/main/java/org/axonframework/axonserver/connector/util/PriorityExecutorService.java

```diff
@@ -0,0 +1,188 @@
+ * is submitted that's an instance of one. The created {@link PriorityTask} will have the priority provided in the
+ * constructor, as well as the taskSequence provided there.
+ * <p>
+ * This implementation also diverts all invocations of {@link ExecutorService#submit(Runnable)} to
+ * {@link ExecutorService#execute(Runnable)} instead. This is because the {@code submit} method implementations wrap the
+ * tasks in another {@link FutureTask}, which makes the task uncomparable again.
+ */
+public class PriorityExecutorService implements ExecutorService {
+
+    private final ExecutorService delegate;
+    private final AtomicLong taskSequence;
+    private final long priority;
+
+    /**
+     * Creates a new {@link PriorityExecutorService} with the provided priority and sequence.
+     *
+     * @param delegate     The delegate {@link ExecutorService} to use when submitting tasks
+     * @param priority     The priority that any tasks submitted to the delegate will have
+     * @param taskSequence The task sequence, used for ordering items within the same priority
+     */
+    PriorityExecutorService(ExecutorService delegate, long priority, AtomicLong taskSequence) {
+        this.priority = priority;
+        this.delegate = delegate;
+        this.taskSequence = taskSequence;
+    }
+
+    @Override
+    public void shutdown() {
+        delegate.shutdown();
+    }
+
+    @Nonnull
+    @Override
+    public List<Runnable> shutdownNow() {
+        return delegate.shutdownNow();
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return delegate.isShutdown();
+    }
+
+    @Override
+    public boolean isTerminated() {
+        return delegate.isTerminated();
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, @Nonnull TimeUnit unit)
+            throws InterruptedException {
+        return delegate.awaitTermination(timeout, unit);
+    }
+
+    private PriorityRunnable createPriorityRunnable(Runnable task) {
+        if (task instanceof PriorityRunnable) {
+            return (PriorityRunnable) task;
+        }
+        return new PriorityRunnable(task, priority, taskSequence.incrementAndGet());
+    }
+
+    private <T> PriorityCallable<T> createPriorityCallable(Callable<T> task) {
+        if (task instanceof PriorityCallable) {
+            return (PriorityCallable<T>) task;
+        }
+        return new PriorityCallable<>(task, priority, taskSequence.incrementAndGet());
+    }
+
+    @Nonnull
+    @Override
+    public <T> Future<T> submit(@Nonnull Callable<T> task) {
+        CompletableFuture<T> future = new CompletableFuture<>();
+        PriorityRunnable priorityTask = createPriorityRunnable(() -> {
+            try {
+                T call = task.call();
+                future.complete(call);
+            } catch (Exception e) {
+                future.completeExceptionally(e);
+            }
+        });
+        delegate.execute(priorityTask);
```

Comment:
It would be nice to explain here why `submit` is delegated to the `execute`.

Reply:
It's in the javadoc on the class